src/libambit/device_driver_ambit3.c

```diff
@@ -95,6 +102,25 @@ ambit_device_driver_t ambit_device_driver_ambit3 = {
     gps_orbit_write
 };
 
+static enum ambit3_fw_gen get_ambit3_fw_gen(ambit_device_info_t *device_info)
+{
+    struct generation {
+        uint8_t fw_version[4];
+        enum ambit3_fw_gen gen;
+    } generations[] = {
+        {{2, 0, 4, 0}, AMBIT3_FW_GEN2},
+        {{0, 0, 0, 0}, AMBIT3_FW_GEN1},
+    };
+    struct generation *generation;
+
+    ARRAY_FOR_EACH(generations, generation) {
+        if (libambit_fw_version_number(generation->fw_version) >= libambit_fw_version_number(device_info->fw_version))
```

Comment:
Syncing moves seems to work with this change with my Ambit 3 Sport with firmware 1.0.16.